File: app/src/main/java/com/bilulo/androidtest04/data/api/statements/StatementsApi.java
package com.bilulo.androidtest04.data.api.statements;

import com.bilulo.androidtest04.data.model.response.StatementsResponse;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface StatementsApi {
    @GET("/api/statements/{userId}")
    Call<StatementsResponse> getStatements(@Path("userId") int userId);
}

File: app/src/main/java/com/farage/testeandroidv2/datasource/api/StatementsApi.kt
package com.farage.testeandroidv2.datasource.api

import com.farage.testeandroidv2.datasource.model.StatementRequest
import kotlinx.coroutines.Deferred
import retrofit2.http.GET
import retrofit2.http.Path

interface StatementsApi {

    @GET("statements/{id}")
    fun getAllStatements(@Path("id") id: Int): Deferred<StatementRequest>

}